ceph_deploy/v_pools.py

```diff
@@ -0,0 +1,243 @@
+import rdbms as model
+import logging
+import json
+from ceph_deploy import hosts
+from ceph_deploy.lib import remoto
+
+# pool tools
+LOG = logging.getLogger(__name__)
+
+def pool_list(conn):
+    stdout, stderr, rc = remoto.process.check(
+        conn,
+        [
+            'ceph',
+            '-f',
+            'json',
+            'osd',
+            'lspools'
+            ],
+        )
+    if rc != 0:
+        LOG.debug("stdout=%s" % (stdout))
+        LOG.debug("stderr=%s" % (stderr))
+        LOG.debug("rc=%s" % (rc))
+        return None
+    auth = json.loads("".join(stdout).strip())
+    return auth
+
+def pool_add(conn, name, number):
+    stdout, stderr, rc = remoto.process.check(
+        conn,
+        [
+            'ceph',
+            'osd',
+            'pool',
+            'create',
+            name,
+            str(number)
+            ],
+        )
+    if rc != 0:
+        LOG.debug("stdout=%s" % (stdout))
+        LOG.debug("stderr=%s" % (stderr))
+        LOG.debug("rc=%s" % (rc))
+        return None
+
+def pool_del(conn, name):
+    stdout, stderr, rc = remoto.process.check(
+        conn,
+        [
+            'ceph',
+            'osd',
+            'pool',
+            'delete',
+            name,
+            name,
+            '--yes-i-really-really-mean-it'
+            ],
+        )
+    if rc != 0:
+        LOG.debug("stdout=%s" % (stdout))
+        LOG.debug("stderr=%s" % (stderr))
+        LOG.debug("rc=%s" % (rc))
+        return None
+
+
+class mdl_updater(object):
```

Comment:
This class and its methods are not very MVC.

Better separation between the Model View's and controller should be done here to be a good example.

I see this code as being split into 2 views one for getting the sate, and one for updating the state of the cluster.

The controller would be elsewhere.